구현_실습/Boilerplate/src/js/ListSection.js

```diff
@@ -0,0 +1,224 @@
+    const wrapperUl = createElement('ul', ListSection.wrapperUlClass);
+    wrapperUl.addEventListener('change', (e) => {
+      if (!(e.target instanceof HTMLInputElement)) {
+        return;
+      }
+      const targetLiEl = e.target.parentNode;
+      const changedStatus = e.target.checked
+        ? StatusSection.STATUS.COMPLETED
+        : StatusSection.STATUS.ACTIVE;
+      if (changedStatus === StatusSection.STATUS.COMPLETED) {
+        // 체크 안된 요소 밑으로 내려가야 함
+        wrapperUl.appendChild(targetLiEl);
+      } else {
+        // 체크 된 요소 위로 올라가야 함
+        wrapperUl.insertBefore(targetLiEl, wrapperUl.firstChild);
+      }
+      this.filterListItem(this.statusInStatusSection, targetLiEl);
+      this.eventObserver?.changeTodoItemStatus.notify(changedStatus);
+    });
+    this.wrapperUl = wrapperUl;
+  }
+
+  get() {
+    return this.wrapperUl;
+  }
+
+  getFromItems(items) {
+    // storybook 용 함수
+    for (const { checked, text } of items) {
+      const newItemLi = this.prependListItem(text);
+      if (checked) {
+        ListSection.getCheckBox(newItemLi).checked = true;
+      }
+    }
+
+    return this.wrapperUl;
+  }
+
+  prependListItem(text) {
+    const newItemLi = this.createListItem(text);
+    if (this.statusInStatusSection === StatusSection.STATUS.COMPLETED) {
+      hideElement(newItemLi);
+    }
+    this.wrapperUl.prepend(newItemLi);
+
+    return newItemLi;
+  }
+
+  createListItem(text) {
+    const inputEl = document.createElement('input');
+    inputEl.type = 'checkbox';
+    const labelEl = document.createElement('label');
+    labelEl.innerText = text;
+
+    const liEl = document.createElement('li');
+    liEl.className = ListSection.itemClass;
+    liEl.dataset.cyId = this.createdItemCnt;
+    this.createdItemCnt = this.createdItemCnt + 1;
+    liEl.appendChild(inputEl);
+    liEl.appendChild(labelEl);
+
+    return liEl;
+  }
+
+  isShowListItemInAll() {
+    return true;
+  }
+
+  isShowListItemInActive(checked) {
+    return !checked;
+  }
+
+  isShowListItemInCompleted(checked) {
+    return checked;
+  }
+
+  isShowListItemByStatus(status, li) {
+    if (status === StatusSection.STATUS.ALL) {
+      return this.isShowListItemInAll();
+    }
+    const { checked } = ListSection.getCheckBox(li);
+    if (status === StatusSection.STATUS.ACTIVE) {
+      return this.isShowListItemInActive(checked);
+    }
+
+    // StatusSection.STATUS.COMPLETED
+    return this.isShowListItemInCompleted(checked);
+  }
+
+  filterListItem(status, li) {
+    if (this.isShowListItemByStatus(status, li)) {
+      showElement(li);
+    } else {
+      hideElement(li);
+    }
+  }
+
+  changeStatusSection(status) {
+    for (const li of this.wrapperUl.childNodes) {
+      this.filterListItem(status, li);
+    }
+    this.statusInStatusSection = status;
+  }
+
+  deleteCompletedItems() {
+    const completedItems = [].filter.call(
+      this.wrapperUl.childNodes,
+      (li) => ListSection.getCheckBox(li).checked
+    );
```

Comment:
`ArrayLike` 객체를 배열로 바꿔주는 `Array.from()`은 어떠신가요? 알아보기 편해서 애용하고 있습니다.

```suggestion
    const completedItems = Array.from(this.wrapperUl.childNodes).filter(
      (li) => ListSection.getCheckBox(li).checked
    );
```